apps/emqx/include/asserts.hrl

```diff
@@ -0,0 +1,42 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
+
+-define(assertWaitEvent(Code, EventMatch, Timeout),
+    ?check_trace(
+        ?wait_async_action(
```

Comment:
Isn't easier to match the return value of this macro?

https://github.com/kafka4beam/snabbkaffe/blob/master/src/snabbkaffe.erl#L190

Wrapping a small piece of code in `?check_trace` (rather than the entire testcase) is problematic, because this macro erases the trace when it finishes. So will cause problems that are extremely hard to troubleshoot when used in combination with other checks that expect the trace to be intact.

Reply:
✅